cmake/Modules/CocosBuildHelpers.cmake

```diff
@@ -75,6 +68,23 @@ function(get_target_depends_ext_dlls cocos_target all_depend_dlls_out)
             break()
         endif()
     endwhile(true)
+    set(${all_depends_out} ${all_depends_inner} PARENT_SCOPE)
+endfunction()
+
+# get `cocos_target` depend all dlls, save the result in `all_depend_dlls_out`
+function(get_target_depends_ext_dlls cocos_target all_depend_dlls_out)
```

Comment:
I think that instead of all those functions and property CC_DEPEND_DLLS set for each target in external libs we should just have list of targets (curl zlib ... ) and used this list where needed

Reply:
Please check
https://github.com/cocos2d/cocos2d-x-3rd-party-libs-bin/issues/330#issuecomment-436728995

I think that we should call function with list of targets
copy_imported_targets(app curl zlib ..

Reply:
> property CC_DEPEND_DLLS set for each target in external libs 

yes, it's proper than this one.

> we should call function with list of targets

I don't think so, since the target lists have existed in the relationship of dependence, we can get them by `LINK_LIBRARIES` property recursively.

The defect is CMake didn't supply a functions to get all linked libraries, so we need `search_depend_libs_recursive `, it's a little complex. About the feature CMake should supply, existed thread at https://gitlab.kitware.com/cmake/cmake/issues/12435